laravel-app/tests/Unit/RegisterControllerTest.php

```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
+    {
+        $user = User::create([
+            'username' => 'Test User',
+            'email' => 'test@example.com',
+            'password' => bcrypt('password'),
+            'email_verification_code' => 'sample_verification_code',
+        ]);
+
+        $verificationUrl = route('verify-email', ['verification_code' => $user->email_verification_code]);
+
+        $response = $this->get($verificationUrl);
+
+        $response->assertRedirect('/login');
+    }
+
+    /** @test */
+    public function test_user_redirected_to_resend_email_if_unverified_email_on_login()
+    {
+        User::create([
+            'username' => 'Unverified User',
+            'email' => 'unverified@example.com',
+            'password' => bcrypt('password'),
+            'email_verified_at' => null,
+        ]);
+
+        $response = $this->post('/login', [
+            'email' => 'unverified@example.com',
+            'password' => 'password',
+        ]);
+
+        $response->assertRedirect('register/resend');
+    }
+
+    /** @test */
+    public function test_user_redirected_to_login_after_resending_email()
+    {
+        $user = User::create([
+            'username' => 'Unverified User',
+            'email' => 'unverified@example.com',
+            'password' => bcrypt('password'),
+            'email_verification_code' => 'sample_verification_code',
+            'email_verified_at' => null,
+        ]);
+
+        $response = $this->post('/login', [
+            'email' => 'unverified@example.com',
+            'password' => 'password',
+        ]);
+
+        $response->assertRedirect('/register/resend');
+
+        $response = $this->get('/register/resend');
+
+        $response->assertViewIs('auth.resend');
+
+        $response = $this->post('/register/resend', [
+            'email-resend' => 'unverified@example.com',
+        ]);
+
+        Mail::assertSent(EmailVerificationMail::class, function ($mail) use ($user) {
+            return $mail->hasTo($user->email);
+        });
+
+        $response->assertRedirect('/login');
+    }
+
+    /** @test */
+    public function test_user_resend_email_error_if_already_verified_or_non_existent_email()
+    {
+        User::create([
+            'username' => 'Verified User',
+            'email' => 'verified@example.com',
+            'password' => bcrypt('password'),
+            'email_verified_at' => now(),
+        ]);
+
+        $responseVerified = $this->post('/register/resend', [
+            'email-resend' => 'verified@example.com',
+        ]);
+
+        $responseVerified->assertSessionHas('error', 'Email doesn\'t exist in the database or is already verified');
+
+        $responseNonExistent = $this->post('/register/resend', [
+            'email-resend' => 'nonexistent@example.com',
+        ]);
+
+        $responseNonExistent->assertSessionHas('error', 'Email doesn\'t exist in the database or is already verified');
+    }
+
```

Comment:
Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

Reply:
Apologies sir 🙇 will revise it now 🙇